### PR TITLE
add koordlet runtime design

### PR DESCRIPTION
**Pull request by zwzhang0107**
Signed-off-by: zwzhang0107 <zuoweizhang@outlook.com>

### Ⅰ. Describe what this PR does
#57

### CODE REVIEW

File: docs/desgin-archive/koordlet-runtime-hooks.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+injection stage from Runtime Manager, such as `PreRunPodSandboxHook` and `PreStartContainerHook`. For each request,
+the gRPC Server will call corresponding plugins to do the injections, fill the protocol parameters 
+and response to Runtime Manager.
+
+#### Rule
+The `Rule` module is responsible for configuration management, parsing the `NodeSLO` changes and update the local cache
+of injection policies. For each policy update, the `Rule` module also provides callback procedure to refresh the
+parameters of pods and containers.  
+
+### Working Mode
```
Comment (hormes) [inline docs/desgin-archive/koordlet-runtime-hooks.md:40]: It is difficult to understand the similarities and differences of the working mechanisms of the two working modes. Can you add a diagram to illustrate the difference between the two modes？

File: docs/desgin-archive/koordlet-runtime-hooks.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+         * [gRPC Server](#grpc-server)
+         * [Rule](#rule)
+      * [Working Mode](#working-mode)
+         * [Proxy](#proxy)
+         * [Bypass](#bypass)
+   * [Examples for hooks extensions](#examples-for-hooks-extensions)
+
+## Summary
+Runtime Hooks acts as the back-end server of Runtime Hook Manager. Runtime Hook Manager is a CRI Proxy, which 
+intercepting the CRI request, calling back-end server to inject policies, such as setting resource isolation
```
Comment (saintube) [inline docs/desgin-archive/koordlet-runtime-hooks.md:19]: typos: `intercepting` -> `intercepts`, `back-end server` -> `the back-end server`

**Review by saintube — COMMENTED** [review]
/lgtm

File: docs/desgin-archive/koordlet-runtime-hooks.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#### gRPC Server
+The `gRPC Server` provides the service for runtime injection, which implements the protocol defined by Runtime Manager.
+After registered the service endpoint to Runtime Manager, the gRPC Server module will receive requests of different 
+injection stage from Runtime Manager, such as `PreRunPodSandboxHook` and `PreStartContainerHook`. For each request,
+the gRPC Server will call corresponding plugins to do the injections, fill the protocol parameters 
+and response to Runtime Manager.
+
+#### Rule
+The `Rule` module is responsible for configuration management, parsing the `NodeSLO` changes and update the local cache
+of injection policies. For each policy update, the `Rule` module also provides callback procedure to refresh the
```
Comment (jasonliu747) [inline docs/desgin-archive/koordlet-runtime-hooks.md:51]: provides `a` callback procedure

File: docs/desgin-archive/koordlet-runtime-hooks.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+#### Bypass
+`Bypass` working mode simplifies the runtime configuration and architecture. `Runtime Manager` is not needed in this 
+working mode, so that `Runtime Hooks` works as an independent module. Without the proxy intercepts requests between 
+Kubelet and Runtime, `Runtime Hooks` can only set related parameters asynchronously, which limits the using scenarios.  
+
+## Examples for hooks extensions
+Group Identity is a kernel feature implemented by Anolis OS, which allows user to configure identities for CPU cgroups
+to prioritize tasks in the cgroups. In the mechanism of Koordlet QoS Management, `LS` pods will get higher identity 
+than `BE` pods. which is implemented as a plugin in `Runtime Hooks` by setting the cgroup parameters
```
Comment (jasonliu747) [inline docs/desgin-archive/koordlet-runtime-hooks.md:74]: than BE pods`,` which is

**Review by jasonliu747 — APPROVED** [review]
/lgtm

File: docs/desgin-archive/koordlet-runtime-hooks.md (hunk 3 above)
Comment (zwzhang0107) [reply docs/desgin-archive/koordlet-runtime-hooks.md]: fixed

File: docs/desgin-archive/koordlet-runtime-hooks.md (hunk 2 above)
Comment (zwzhang0107) [reply docs/desgin-archive/koordlet-runtime-hooks.md]: fixed

**Pull request merged**